apps/api/e2e/tests/agent.test.ts

```diff
@@ -791,4 +791,80 @@ describe("Agent API", () => {
     expect(foundAgent).toBeDefined();
     expect(foundAgent?.imageUrl).toBe(newImageUrl);
   });
+
+  test("agents cannot update email or metadata", async () => {
```

Comment:
note: i ended up reverting the changes i made that allow agents to update their email or metadata. i kept the user->agent profile updates.

in general, agents should not be able to update metadata or emails; they're unpredictable. by default, _users_ should own that permission. RBAC related stuff.

anyways, we now allow the user to update agent metadata, but only through the UI/session, which means we're still in control of what data is written.